resources/Datastore.go

```diff
@@ -0,0 +1,162 @@
+package resources
+
+import "github.com/beevik/etree"
+
+// Datastore structure represents OpenNebula datastore
+type Datastore struct {
+	Resource
+}
+
+// DatastoreTypeMap contains string representation of DatastoreType
+var DatastoreTypeMap = map[DatastoreType]string{
+	ImageDs:  "IMAGE_DS",
+	SystemDs: "SYSTEM_DS",
+	FileDs:   "FILE_DS",
+}
+
+// DatastoreType - type of datastore
+type DatastoreType int
+
+const (
+	// ImageDs - Standard datastore for disk images
+	ImageDs DatastoreType = iota
+	// SystemDs - System datastore for disks of running VMs
+	SystemDs
+	// FileDs - File datastore for context, kernel, initrd files
+	FileDs
+)
+
+// DatastoreDiskTypeMap contains string representation of DatastoreDiskType
+var DatastoreDiskTypeMap = map[DatastoreDiskType]string{
+	File:          "FILE",
```

Comment:
Since Go doesn't use any useable form of namespacing for constants (I checked. For example take a look at `net/http` source code an [constant naming](https://golang.org/pkg/net/http/#pkg-constants) there) I would prefer prefixing/suffixing all the constants so it's obvious what does it stands for from its name.